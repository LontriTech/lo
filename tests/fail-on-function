#!/usr/bin/env bash
cd "${0%/*}" || exit 1
. ../run-tests

failFunction() {
    false
}
arguments=(--fail-on-function)
returnCode=1

template="Fail on function? {{failFunction}}"
expected() {
    cat <<EOF
ERROR: Function 'failFunction' with args () failed with status code 1
EOF
}

runTest
