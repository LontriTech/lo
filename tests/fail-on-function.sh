#!/usr/bin/env bash

cd "${0%/*}" || exit 1

failFunction() {
    false
}

# Must be sourced to use functions
# shellcheck disable=SC1091
. ../mo
mo --fail-on-function 2>&1 <<EOF
Fail on function? {{failFunction}}
EOF

if [[ $? -ne 1 ]]; then
    echo "Did not return 1"
fi
