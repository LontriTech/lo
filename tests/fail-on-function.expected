Fail on function? Function 'failFunction' with args () failed with status code 1
